### PR TITLE
Update the types consumers of Index and Range

### PR DESCRIPTION
**Pull request by tarekgh**
The C# compiler now can automatically support the indexers with Index and Range parameters on any type meet the conditions:
-	The type exposes Length or Count property which returning integer.
-	The type has indexer of integer parameter.
-	The type has a Slice method which takes 2 integers

The change here is to remove the methods and indexers of the types which we previously added it as the compiler will automatically provide those.

### CODE REVIEW

**Comment by tarekgh** [comment]
CC @stephentoub @agocke @terrajobst @GrabYourPitchforks @joshfree @jkotas

File: src/System.Private.CoreLib/src/System/Utf8String.Manipulation.cs [hunk 1]
```diff
@@ -77,11 +77,9 @@ public Utf8String Substring(int startIndex, int length)
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public Utf8String Substring(Range range)
-        {
-            (int start, int length) = range.GetOffsetAndLength(Length);
-            return Substring(start, length);
-        }
+        [ComponentModel.EditorBrowsable(ComponentModel.EditorBrowsableState.Never)]
+        public Utf8String Slice(int startIndex, int length) => Substring(startIndex, length);
```
Comment (GrabYourPitchforks) [inline src/System.Private.CoreLib/src/System/Utf8String.Manipulation.cs]: Should probably have a comment that we don't intend for developers to call this method directly; it's pattern-matched by the compiler to provide indexer functionality.

Comment (tarekgh) [reply src/System.Private.CoreLib/src/System/Utf8String.Manipulation.cs]: I'll do that.

**Comment by tarekgh** [comment]
CC @AndyAyersMS as some of jit tests failed because of this change. Should we delete these tests? 

https://mc.dot.net/#/user/coreclr-ci/pr~2Fdotnet~2Fcoreclr~2Frefs~2Fpull~2F24036~2Fmerge/test~2Ffunctional~2Fcli~2F/20190416.713/workItem/JIT.Regression/analysis/xunit/JIT_Regression._JitBlue_GitHub_20958_GitHub_20958_GitHub_20958_~2F_JitBlue_GitHub_20958_GitHub_20958_GitHub_20958_sh

```
Unhandled Exception: System.MissingMethodException: Method not found: 'System.ReadOnlySpan`1<!0> System.ReadOnlySpan`1.get_Item(System.Range)'.
   at GitHub_20958.IndexerWithRangeTest()
   at GitHub_20958.Main()
/Users/dotnet-bot/dotnetbuild/work/f4214b88-e840-4020-a6d0-f27d59dc2d39/Work/8af4e13c-d569-44cd-a73e-04cf67f18258/Exec/JIT/Regression/JitBlue/GitHub_20958/GitHub_20958/GitHub_20958.sh: line 252:   691 Abort trap: 6           $LAUNCHER $ExePath \"${CLRTestExecutionArguments[@]}\"

```

**Comment by AndyAyersMS** [comment]
Deleting the tests is ok, the regression they were guarding against can't happen if there's just an int indexer (and the jit has been fixed anyways).

I take it we don't have the new compiler version available when building the tests?

**Comment by tarekgh** [comment]
> I take it we don't have the new compiler version available when building the tests?

I recall you mentioned before these tests built against the packages and not against  the live bits. Which means, the code was binding to a real Span[Range] indexer. Then the test runs against the new bits which cannot find such indexer because it is not there. 
@agocke may confirm my theory here.

**Comment by agocke** [comment]
That theory sounds right, because the compiler shouldn't ever be able to emit a call to a method that doesn't exist. We wouldn't even have a signature to emit. MissingMethod should basically only ever happen with a gap between the reference and the runtime, or the usage of some dynamic dispatch.

**Comment by tarekgh** [comment]
Test coreclr-ci (Test Pri0 Linux_musl x64 release)

https://mc.dot.net/#/user/coreclr-ci/pr~2Fdotnet~2Fcoreclr~2Frefs~2Fpull~2F24036~2Fmerge/test~2Ffunctional~2Fcli~2F/20190416.726/workItem/JIT/wilogs

```
Executed on a001RRO using docker image mcr.microsoft.com/dotnet-buildtools/prereqs:alpine-3.8-helix-45b1fa2-20190327215821
running $HELIX_CORRELATION_PAYLOAD/scripts/740caaa4040446438ec7c8b99b6f7c16/execute.sh in /home/helixbot/work/b87ffa7d-fd10-4e06-822b-abee43866057/Work/706eb2de-615d-4358-8bb2-98bcb50b4e5e/Exec max 1800 seconds

Output:
Unable to pull image mcr.microsoft.com/dotnet-buildtools/prereqs:alpine-3.8-helix-45b1fa2-20190327215821
Exit Code:-4
```

**Comment by joperezr** [comment]
@dotnet-build-bot test Windows_NT x86 full_opt ryujit CoreCLR Perf Tests Correctness please

**Comment by MattGal** [comment]
@tarekgh the issue you're pointing out is https://github.com/dotnet/core-eng/issues/5883;  we've holding off on rollouts right now, but will hopefully be able to do so soon.

**Comment by tarekgh** [comment]
test Ubuntu arm Cross Checked crossgen_comparison Build and Test please
test Ubuntu arm Cross Release crossgen_comparison Build and Test please

**Pull request merged**